pkg/metrics/experimental.go

```diff
@@ -0,0 +1,40 @@
+// +build experimental_metrics
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	// Register experimental metrics
+	reconcileMetrics = reconcileCounters(operatorController, adoptionCSVController, adoptionSubscriptionController, operatorConditionController, operatorConditionGeneratorController)
+	registerReconcileMetrics()
+	fmt.Printf("metrics")
+}
+
+func reconcileCounters(reconcilerNames ...string) map[string]prometheus.Counter {
+	result := map[string]prometheus.Counter{}
+	for _, s := range reconcilerNames {
+		result[s] = reconcileCounter(s)
+	}
+	return result
+}
+
+func reconcileCounter(name string) prometheus.Counter {
+	return prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "controller_reconcile_" + name,
+			Help: fmt.Sprintf("Count of reconcile events by the %s controller", strings.Replace(name, "_", " ", -1)),
```

Comment:
Would you be able to provide some context as to why these types of reconciliation counter gauges are useful? Anything in particular you had in mind that the introduction of these metrics help improve?

Reply:
Primary value add is the ability to identify when a controller hoploops / if it is capable of doing so. I need to change this to a counterVec so we can capture which resource is triggering the hotloop.

Reply:
Awesome, yeah that makes a lot more sense to me. This should be really helpful for identifying issues long term through CI.